src/main/java/com/linecorp/armeria/server/ServiceMapping.java

```diff
@@ -111,9 +111,7 @@ public MappedService apply(String path) {
 
         // Cache miss or disabled cache
         MappedService result = MappedService.empty();
-        final int size = patterns.size();
-        for (int i = 0; i < size; i ++) {
-            final Entry<PathMapping, Service> e = patterns.get(i);
+        for (final Entry<PathMapping, Service> e : patterns) {
```

Comment:
Everything is a trade off. Just for the record this benchmark http://blog.takipi.com/benchmark-how-java-8-lambdas-and-streams-can-make-your-code-5-times-slower/ shows a performance difference between the two styles. Without a benchmark as a reason to go with a hand rolled optimization we should stick with readability !